pqf/research/statistics.py

```diff
@@ -40,6 +40,67 @@ def sharpe_ratio(
     return sharpe
 
 
+def sortino_ratio(
+    returns: pl.Series | pl.Expr, risk_free_rate: float
+) -> float | pl.Expr | None:
+    """Calculate the Sortino Ratio.
+
+    Args:
+        returns (pl.Series | pl.Expr): Series or expression representing returns.
+        risk_free_rate (float): The risk-free rate.
+
+    Raises:
+        TypeError: If downside deviation cannot be calculated.
+
+    Returns:
+        float | pl.Expr | None: The calculated Sortino ratio or None if not calculable.
+    """
+    if isinstance(returns, pl.Series):
+        mean_return = returns.mean()
+        if mean_return is None or not isinstance(mean_return, (float, int)):
+            return None
+
+        mean_excess_return = mean_return - risk_free_rate
+
+        negative_returns = returns.filter(returns < 0)
+        if negative_returns.is_empty():
+            return None
+
+        downside_deviation = negative_returns.std()
+        if downside_deviation is None:
+            raise TypeError(
+                "Downside deviation must be a numeric type and could not be calculated."
+            )
+
+        # Explicitly cast downside_deviation to float if necessary
+        downside_deviation_float = (
```

Comment:
This seems redundant, why do we have to ensure that it is a float? An integer should be fine?